### ISSUE TITLE
qxmpp 1.5.0 build failure

### ISSUE THREAD

**Issue by chenrui333** [qxmpp-project/qxmpp#531]
👋 trying to build the latest release, but run into some build issue. The error log is as below:

<details>
<summary>error build log</summary>

```
/tmp/qxmpp-20230131-90264-1ofa8bi/qxmpp-1.5.0/src/client/QXmppClient.cpp:415:49: error: 'auto' not allowed in template argument
    const auto sendEncrypted = [this](QXmppTask<auto> &&task) {
                                                ^~~~
/tmp/qxmpp-20230131-90264-1ofa8bi/qxmpp-1.5.0/src/client/QXmppClient.cpp:440:20: error: no matching function for call to object of type 'const (lambda at /tmp/qxmpp-20230131-90264-1ofa8bi/qxmpp-1.5.0/src/client/QXmppClient.cpp:415:32)'
            return sendEncrypted(
                   ^~~~~~~~~~~~~
/tmp/qxmpp-20230131-90264-1ofa8bi/qxmpp-1.5.0/src/client/QXmppClient.cpp:415:32: note: candidate function not viable: no known conversion from 'QXmppTask<QXmppE2eeExtension::MessageEncryptResult>' (aka 'QXmppTask<variant<std::unique_ptr<QXmppMessage>, QXmppError>>') to 'int' for 1st argument
    const auto sendEncrypted = [this](QXmppTask<auto> &&task) {
                               ^
/tmp/qxmpp-20230131-90264-1ofa8bi/qxmpp-1.5.0/src/client/QXmppClient.cpp:444:20: error: no matching function for call to object of type 'const (lambda at /tmp/qxmpp-20230131-90264-1ofa8bi/qxmpp-1.5.0/src/client/QXmppClient.cpp:415:32)'
            return sendEncrypted(
                   ^~~~~~~~~~~~~
/tmp/qxmpp-20230131-90264-1ofa8bi/qxmpp-1.5.0/src/client/QXmppClient.cpp:415:32: note: candidate function not viable: no known conversion from 'QXmppTask<QXmppE2eeExtension::IqEncryptResult>' (aka 'QXmppTask<variant<std::unique_ptr<QXmppIq>, QXmppError>>') to 'int' for 1st argument
    const auto sendEncrypted = [this](QXmppTask<auto> &&task) {
                               ^
```

</details>

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/4059822619/jobs/6988371038
relates to Homebrew/homebrew-core#121997

**Comment by chenrui333** [comment]
I am building against qt 6.4.2

**Comment by lnjX** [comment]
Thanks for testing, fixed in e2710926c637ab80efbada2127a32512cf2f545c, also see https://github.com/qxmpp-project/qxmpp/releases/tag/v1.5.1.

**Comment by chenrui333** [comment]
confirmed the fix in [1.5.1 build](https://github.com/Homebrew/homebrew-core/pull/122348), thanks @lnjX!
